tapscript/tx.go

```diff
@@ -99,125 +186,146 @@ func virtualTxIn(newAsset *asset.Asset, prevAssets commitment.InputSet) (
 		return nil, nil, ErrInputMismatch
 	}
 
-	treeRoot, err := inputTree.Root(context.Background())
-	if err != nil {
-		return nil, nil, err
-	}
+	return txIns, aggrTree, nil
+}
 
-	// TODO(roasbeef): document empty hash usage here
-	prevOut := asset.VirtualTxInPrevOut(treeRoot)
+// virtualTxOut computes the outputs of a Taproot Asset virtual transaction.
+// Each output's pkScript commits to an MS-SMT containing the asset output at
+// the same index.
+func virtualTxOut(assetOutputs []*asset.Asset) ([]*wire.TxOut, error) {
+	var (
+		txOuts = make([]*wire.TxOut, 0, len(assetOutputs))
+	)
 
-	return wire.NewTxIn(prevOut, nil, nil), inputTree, nil
-}
+	for _, txAsset := range assetOutputs {
+		// If we have any asset splits, then we'll indirectly commit to
+		// all of them through the SplitCommitmentRoot.
+		if txAsset.SplitCommitmentRoot != nil {
```

Comment:
This can prove problematic for some sighash types. As this output implicitly commits to all outputs, altering the desired effect of the sighash

Reply:
Yeah, you're right. I think we won't be able to support the `SIGHASH_SINGLE` use case on the asset level. Because when you create the outputs, you need to know _all_ outputs, otherwise you can't create a split commitment.
But looking at the example given in https://github.com/lightninglabs/taproot-assets/issues/577 it seems like we mostly want the `SIGHASH_NONE` functionality (at least on the output side). So that should still be achievable.

Reply:
> Yeah, you're right. I think we won't be able to support the SIGHASH_SINGLE use case on the asset level.

Agreed. Though I think it doesn't matter, as the root bitcoin signature is what actually binds everything. See my comment above. We don't need to enumerate the outputs as the split does that for us already.